src/go/configgenerator/route_generator_test.go

```diff
@@ -21,13 +21,84 @@ import (
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/configinfo"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/proto"
 
 	routepb "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
 	matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
 	wrapperspb "github.com/golang/protobuf/ptypes/wrappers"
 )
 
+func TestMakeRouteConfig(t *testing.T) {
+	testData := []struct {
+		desc                          string
+		enableStrictTransportSecurity bool
+		wantedError                   string
+		wantRouteConfig               string
+	}{
+		{
+			desc:                          "Enable Strict Transport Security",
+			enableStrictTransportSecurity: true,
+			wantRouteConfig: `{
+                             "name": "local_route",
+                             "virtualHosts": [
+                                 {
+                                     "domains": [
+                                         "*"
+                                     ],
+                                     "name": "backend",
+                                     "routes": [
+                                         {
+                                             "match": {
+                                                 "prefix": "/"
```

Comment:
why not making this for all backend?  why just remote backend?  As I see it,  HSTS is per server flag,  proxy is one listener is one server, this flag should apply to all backend the proxy is supporting.

I understand that it is only useful for side-car (local host) backend, which usually is connected using http.  so Backend will not respond with HSTS header. Remote backend usually is https, it is up to each https backend to response such header. 

But I see such flag is a per-server flag, it should apply to ESPv2 proxy as a whole.
It also help with Cloud Run for anthos which is using http remote backend.

Reply:
Added.